### PR TITLE
Login Screen Notice: view plaintext notice

### PR DESCRIPTION
**Pull request by SammyIsConfused**
#660

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/login/LogonSection.java [hunk 1]
```diff
@@ -252,6 +252,10 @@ public SectionResult renderHtml(RenderEventContext context)
 		Decorations decorations = Decorations.getDecorations(context);
 		decorations.setTitle(TITLE_LABEL);
 		decorations.setMenuMode(MenuMode.HIDDEN);
+		String loginNotice = loginNoticeService.getPreLoginNotice();
+
+		if (loginNotice!=null&&!loginNotice.isEmpty())
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/login/LogonSection.java]: * We'd want space around those operators;
* We'd also want to wrap the branch in braces for clarity; but
* The guard is probably redundant - as I see normally this is working with `null` when it's not set - so you could maybe just go `model.setLoginNotice( loginNoticeService.getPreLoginNotice() );`

**Comment by SammyIsConfused** [comment]
New UI: 
image
Old UI:
image

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/login/LogonSection.java (hunk 1 above)
Comment (SammyIsConfused) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/login/LogonSection.java]: e73e924

File: Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java [hunk 2]
```diff
@@ -48,8 +48,12 @@ public String getPreLoginNotice()
 	@Override
 	public void setPreLoginNotice(String notice)
 	{
-		checkPermissions();
-		configurationService.setProperty(PRE_LOGIN_NOTICE_KEY, notice);
+		if(notice.equals("")) {
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java]: Check for null.

File: Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java [hunk 3]
```diff
@@ -48,8 +48,12 @@ public String getPreLoginNotice()
...
+			deletePreLoginNotice();
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java]: Let's change this to use the `configurationService.deleteProperty`, and then we can have a clear single use of `checkPermissions` at the start of this method.

File: Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java [hunk 4]
```diff
@@ -68,8 +72,12 @@ public String getPostLoginNotice()
...
 	public void setPostLoginNotice(String notice)
 	{
-		checkPermissions();
-		configurationService.setProperty(POST_LOGIN_NOTICE_KEY, notice);
+		if(notice.equals("")){
+			deletePostLoginNotice();
+		}else{
+			checkPermissions();
+			configurationService.setProperty(POST_LOGIN_NOTICE_KEY, notice);
+		}
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java]: Repeat comments above.

File: Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java [hunk 5]
```diff
@@ -49,7 +50,11 @@ public String getPreLoginNotice()
 	public void setPreLoginNotice(String notice)
 	{
 		checkPermissions();
-		configurationService.setProperty(PRE_LOGIN_NOTICE_KEY, notice);
+		if(StringUtils.isEmpty(notice)) {
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java]: Use `isBlank` instead.

File: Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java [hunk 6]
```diff
@@ -69,7 +74,11 @@ public String getPostLoginNotice()
 	public void setPostLoginNotice(String notice)
 	{
 		checkPermissions();
-		configurationService.setProperty(POST_LOGIN_NOTICE_KEY, notice);
+		if(StringUtils.isEmpty(notice)) {
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/loginnotice/impl/LoginNoticeServiceImpl.java]: Use `isBlank` instead.

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/login/LogonSection.java [hunk 7]
```diff
@@ -252,6 +252,8 @@ public SectionResult renderHtml(RenderEventContext context)
 		Decorations decorations = Decorations.getDecorations(context);
 		decorations.setTitle(TITLE_LABEL);
 		decorations.setMenuMode(MenuMode.HIDDEN);
+		String loginNotice = loginNoticeService.getPreLoginNotice();
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/login/LogonSection.java]: Variable is redundant.

**Pull request merged**